Setup frontend

saicheongkam opened the pull request.

Commit:
Setup Unit Testing

Setup QUnit Unit Testing with some examples using dummy Participant and
Enterprise data.

Commit:
Revert "Setup Unit Testing"

This reverts commit 0ddbb82e44e71690ab39ddf41f93bd6313b8dd83.

Commit:
Setup Unit Testing

Created a few QUnit Unit Tests using dummy Participant and Enterprise
data.

danielelkington commented on Frontend/app/tests/example.ts:
```diff
@@ -0,0 +1,31 @@
+  assert.equal( 0, false, "Zero, false; equal succeeds" );
+});*/
+
+var participants = require("../pages/enterprises/participants.js");						//Access Participant variables and functions
```
Given that we don't have a pages folder, let alone a participants file, how does this work?

saicheongkam replied:
My mistake. I did testing with the existing code we wrote for the Prototype to see how it can be used to access variables and functions from other files. I should have left it out since we haven't done any of this for the main app yet.

danielelkington commented on Frontend/app/tests/example.ts:
```diff
@@ -0,0 +1,31 @@
+var participant2 = new participants.Participant("Kevin11");
+var participant3 = new participants.Participant("Bob12");
+var participantList = [participant1, participant2, participant3];
+var enterprises = require("../pages/enterprises/enterprise.js");
```
Same as above - as far as I can tell the enterprise class isn't defined anywhere, so how would this work? Ideally at all stages the Master branch should compile and all tests should pass.

saicheongkam replied:
Oops! This works for the prototype, but not for the main app at this stage.

danielelkington reviewed (changes requested):
For the moment given that there isn't anything to test in our code I'd be happy to have a single test asserting that 1+1 = 2. Is adding this file really all we need to do to make unit testing work? I'm surprised that there's no packages or plugins that need to be added...

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into setup-frontend

Commit:
Removed tests on non-existing files

Commit:
Add QUnit Framework

Commit:
Changes to make running tests work
Most important change seemed to be updating gitignore so I could add karma.conf.js

danielelkington reviewed (approved):
I still couldn't get this to work on my machine, but on looking into it I think the problem was that our .gitignore ignores all .js files, and karma.conf.js is a critical file needed for unit testing. I've updated .gitignore to include this file, added that file, and (along with a couple of other changes) it seems to be working now. Happy for you to merge. 👍
